nilearn/mass_univariate/_utils.py

```diff
@@ -0,0 +1,80 @@
+"""Utility functions for the permuted least squares method."""
+import nibabel as nib
+import numpy as np
+from scipy import ndimage
+from nilearn.masking import apply_mask, unmask
+
+
+def _calculate_tfce(scores_array, masker, E=0.5, H=2, dh=0.1):
```

Comment:
Here is the relevant description from [Smith & Nichols (2009)](https://pubmed.ncbi.nlm.nih.gov/18501637/):

>Each voxel's TFCE score is given by the sum of the scores of all “supporting sections” underneath it; as the height h is incrementally raised from zero up to the height (signal intensity) _hp_ of a given point _p_, the image is thresholded at _h_, and the single contiguous cluster containing _p_ is used to define the score for that height _h_. This score is simply the height _h_ (raised to some power _H_) multiplied by the cluster extent _e_ (raised to some power _E_). Precisely, the TFCE output at voxel _p_ is
> ![image](https://user-images.githubusercontent.com/8228902/160427721-12dea754-7eb2-4b15-b099-5cc2c5b00635.png)
>where _h0_ will typically be zero, and _E_ and _H_, 0.5 and 2 respectively (these choices are discussed in detail below). In practice this integral is estimated as a sum, using finite _dh_ (for example, _dh_ = 0.1 if the input is a raw _t_ or Z image).

Reply:
Great! Thanks, this is super useful! 👍

Reply:
Done!